cmake/sysbuild/bootloader_dts_utils.cmake

```diff
@@ -0,0 +1,146 @@
+  sysbuild_dt_reg_addr(partition_offset IMAGE ${DEFAULT_IMAGE} PATH ${partition_node})
+
+  # Get the parent "two levels up" (../../) of the partition node
+  # This is the partition flash area node
+  string(REPLACE "/" ";" partition_node_split ${partition_node})
+  list(LENGTH partition_node_split child_path_length)
+  math(EXPR parent_path_length "${child_path_length} - 2")
+  list(SUBLIST partition_node_split 0 ${parent_path_length} parent_path_split)
+  string(REPLACE ";" "/" flash_area_node "${parent_path_split}")
+
+  sysbuild_dt_reg_addr(flash_area_addr IMAGE ${DEFAULT_IMAGE} PATH ${flash_area_node})
+
+  math(EXPR ${address} "${flash_area_addr} + ${partition_offset}")
+  set(${address} ${${address}} PARENT_SCOPE)
+endfunction()
+
+#
+# Verify that the code partition for a given image matches a specific node label.
+#
+function(code_partition_matches_label_verify label image)
+  sysbuild_dt_nodelabel(label_node IMAGE ${image} NODELABEL ${label})
+  sysbuild_dt_chosen(code_partition_node IMAGE ${image} PROPERTY "zephyr,code-partition")
+
+  if (NOT "${code_partition_node}" STREQUAL "${label_node}")
+    message(FATAL_ERROR "
+      ERROR: zephyr,code-partition for image ${image}
+      does not match label ${label}.
+      This is required by the current bootloader configuration.
+      zephyr,code-partition node:
+      ${code_partition_node}
+      ${label} node:
+      ${label_node}
+      \n"
+    )
+  endif()
+endfunction()
+
+#
+# Verify that the required node labels for all images point to the same nodes
+# as for the default image
+#
+function(labels_match_for_all_images_verify labels)
+  foreach(label ${labels})
+    sysbuild_dt_nodelabel(default_image_node IMAGE ${DEFAULT_IMAGE} NODELABEL ${label} REQUIRED)
+    foreach(image ${IMAGES})
+      sysbuild_dt_nodelabel(node IMAGE ${image} NODELABEL ${label} REQUIRED)
+      if (NOT "${node}" STREQUAL "${default_image_node}")
+        message(FATAL_ERROR "
+          ERROR: Node pointed by ${label} for image ${image}
+          does not match the node for the default image ${DEFAULT_IMAGE}.
+          Node for default image ${DEFAULT_IMAGE}: ${default_image_node}
+          Node for image ${image}: ${label_node}
+          \n"
+        )
+      endif()
+    endforeach()
+  endforeach()
+endfunction()
+
+
+function(verify_bootloader_dts_configuration)
+  set(required_labels_list)
+  set(nordic_bootloader_vars_names)
+
+  # Create a list of all devicetree labels that are required for
+  # the current bootloader configuration.
+  list(APPEND required_labels_list "slot0_partition")
+  if(SB_CONFIG_SECURE_BOOT)
+    list(APPEND required_labels_list "b0_partition" "provision_partition")
+  endif()
+  if(SB_CONFIG_BOOTLOADER_MCUBOOT OR SB_CONFIG_SECURE_BOOT_BUILD_S1_VARIANT_IMAGE)
+    # Slot 1 will be used either for the MCUBoot secondary image or for the S1 variant image
+    # The latter will be the case if MCUBOOT is not present
+    list(APPEND required_labels_list "slot1_partition")
```

Comment:
hmm don't like this, slot0/slot1 are for MCUboot, should be using s0/s1 for b0, not mixing it up between both if MCUboot is enabled or disabled

Reply:
Sure, I also was unsure about it, see now.